### PR TITLE
Item movers improvements

### LINKED ISSUE #8236: Can't move item command in some circumstances
[Move item command](https://github.com/rust-analyzer/rust-analyzer/pull/8054) was recently added to rust-analyzer.

```rust
|#[derive(Debug)]
enum FooBar {
    Foo,
    Bar,
}
```

If the cursor is on the left side of the `#` symbol, the move item action cannot be performed. Same goes for the keywords `fn` or `impl` (or an `enum` without an attribute above it).

```rust
|const FOO: &str = "Bar";
```

With the cursor on the left side of a `const` however, the move action works without needing to "be inside" it (i.e. cursor to the right side of `c`).

```rust
fn foobar() {
    // ...
}|
```

Interestingly, if the cursor is on the right side of the ending curly brace of a function (what could be considered "not inside" the function), the move action still succeeds.

I don't know if this all expected behavior, but I thought I'd mention it as it seems inconsistent.

Thanks.

rust-analyzer nightly https://github.com/rust-analyzer/rust-analyzer/commit/01dc53a31c39fe754c7228646c456325e49e8d09 (2021-03-29)

### PR DESCRIPTION
**Pull request by ivan770**
Closes #8236

### CODE REVIEW

File: crates/ide/src/move_item.rs [hunk 1]
```diff
@@ -121,6 +141,17 @@ fn replace_nodes(first: &SyntaxNode, second: &SyntaxNode) -> TextEdit {
     edit.finish()
 }
 
+fn pick_best(tokens: TokenAtOffset<SyntaxToken>) -> Option<SyntaxToken> {
+    return tokens.max_by_key(priority);
+    fn priority(n: &SyntaxToken) -> usize {
```
Comment (matklad) [inline crates/ide/src/move_item.rs:146]:
```suggestion
    return tokens.max_by_key(priority);
    
    fn priority(n: &SyntaxToken) -> usize {
```

**Review by matklad — COMMENTED** [review]
lgtm!

bors d+

File: crates/ide/src/move_item.rs (hunk 1 above)
Comment (ivan770) [reply crates/ide/src/move_item.rs:146]: I'll apply fix to https://github.com/rust-analyzer/rust-analyzer/blob/563464bfaeec9e39de52cd12c435c299fe544b71/crates/ide/src/hover.rs#L471-L475 too

**Comment by lnicola** [comment]
changelog fix make "Move item" commands work in more cases.

**Pull request merged**